api-v2/src/student/student.service.ts

```diff
@@ -0,0 +1,108 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import {
+  DeleteResult,
+  FindOneOptions,
+  Repository,
+  UpdateResult,
+} from 'typeorm';
+import { CreateStudentDto } from './dto/create-student.dto';
+import { UpdateStudentDto } from './dto/update-student.dto';
+import { Student } from './entities/student.entity';
+import { StudentNotFoundError } from './student-not-found.error';
+
+@Injectable()
+export class StudentService {
+  constructor(
+    @InjectRepository(Student)
```

Comment:
with above, this would then just be [this](https://github.com/sandboxnu/office-hours/blob/cca52eb54092abe674fb53f3086453bc61613b94/packages/server/src/course/course.service.ts#L29) – fewer params/code, but less specific. up to you